Libraries/LibJS/Runtime/ArrayBuffer.cpp

```diff
@@ -112,6 +112,13 @@ void copy_data_block_bytes(ByteBuffer& to_block, u64 to_index, ByteBuffer const&
     // 5. Assert: toIndex + count ≤ toSize.
     VERIFY(to_index + count <= to_size);
 
+    // OPTIMIZATION: If neither block is a Shared Data Block, we can use memcpy.
+    auto can_use_fast_path = true;
+    if (can_use_fast_path) {
```

Comment:
Well this doesn't seem like the right condition 😅

Reply:
Replaced with an unconditional branch

Reply:
Oh right - I misunderstood your intention there!